lib/client/api.go

```diff
@@ -3269,6 +3267,14 @@ func (tc *TeleportClient) Login(ctx context.Context) (*Key, error) {
 	// Perform the ALPN test once at login.
 	tc.TLSRoutingConnUpgradeRequired = client.IsALPNConnUpgradeRequired(ctx, tc.WebProxyAddr, tc.InsecureSkipVerify)
 
+	if tc.KeyTTL == 0 {
+		tc.KeyTTL = time.Duration(pr.Auth.DefaultSessionTTL)
+	}
```

Comment:
Looking at the code around it, I think fixing this might be a matter of adding the changes from this file to [`SyncAuthPreference`](https://github.com/gravitational/teleport/blob/274db4a3f908b8bbe0ae578c67c80e6bce5c418f/lib/teleterm/clusters/cluster_auth.go#L40).

Too bad that the auth code is copied over and we don't have tests which would catch regressions.